ui/tests/acceptance/settings/mount-secret-backend-test.js

```diff
@@ -301,26 +314,123 @@ module('Acceptance | settings/mount-secret-backend', function (hooks) {
     );
   });
 
-  test('it sets identity_token_key when aws is chosen, resets after', async function (assert) {
-    // create an oidc/key
-    await runCmd(`write identity/oidc/key/some-key allowed_client_ids="*"`);
-    await page.visit();
-    await page.selectType('aws');
-    await click('[data-test-toggle-group="Method Options"]');
-
-    assert.dom('[data-test-search-select-with-modal]').exists('Search select with modal component renders');
-
-    await clickTrigger();
-    const dropdownOptions = findAll('[data-option-index]').map((o) => o.innerText);
-    assert.ok(dropdownOptions.includes('some-key'), 'search select options show some-key');
-    // Go back and choose a non-wif engine type
-    await page.back();
-    await page.selectType('ssh');
-    await click('[data-test-toggle-group="Method Options"]');
-    assert
-      .dom('[data-test-search-select-with-modal]')
-      .doesNotExist('for type ssh, the modal field does not render.');
-    // clean up
-    await runCmd(`delete identity/oidc/key/some-key`);
+  module('WIF secret engines', function () {
+    test('it sets identity_token_key on mount config using search select list, resets after', async function (assert) {
+      // create an oidc/key
+      await runCmd(`write identity/oidc/key/some-key allowed_client_ids="*"`);
+
+      for (const engine of WIF_SECRET_ENGINES) {
+        await page.visit();
+        await page.selectType(engine);
+        await click(GENERAL.toggleGroup('Method Options'));
+        assert
+          .dom('[data-test-search-select-with-modal]')
+          .exists('Search select with modal component renders');
+        await clickTrigger('#key');
+        const dropdownOptions = findAll('[data-option-index]').map((o) => o.innerText);
+        assert.ok(dropdownOptions.includes('some-key'), 'search select options show some-key');
+        await click(GENERAL.searchSelect.option(GENERAL.searchSelect.optionIndex('some-key')));
+        assert
+          .dom(GENERAL.searchSelect.selectedOption())
+          .hasText('some-key', 'some-key was selected and displays in the search select');
+      }
+      // Go back and choose a non-wif engine type
+      await page.back();
+      await page.selectType('ssh');
+      +assert
```

Comment:
it looks like there's a typo here? 🔍 
```suggestion
      assert
```